src/main/java/uk/ac/cam/cl/dtg/segue/etl/ContentIndexer.java

```diff
@@ -671,10 +673,21 @@ private boolean checkForContentErrors(final String sha, final Map<String, Conten
             if (c instanceof Media) {
                 Media f = (Media) c;
 
-                if (f.getSrc() != null
-                        && !f.getSrc().startsWith("http") && !database.verifyGitObject(sha, f.getSrc())) {
-                    this.registerContentProblem(c, "Unable to find Image: " + f.getSrc()
-                            + " in Git. Could the reference be incorrect? SourceFile is " + c.getCanonicalSourceFile(), indexProblemCache);
+                if (f.getSrc() != null && !f.getSrc().startsWith("http")) {
+                    ByteArrayOutputStream fileData = null;
+                    try {
+                        fileData = database.getFileByCommitSHA(sha, f.getSrc());
+                    } catch (IOException | UnsupportedOperationException e) {
+                        // Leave fileData = null;
+                    }
+                    if (null == fileData) {
+                        this.registerContentProblem(c, "Unable to find Image: " + f.getSrc()
+                                + " in Git. Could the reference be incorrect? SourceFile is " + c.getCanonicalSourceFile(), indexProblemCache);
```

Comment:
Looks good - just wondering why you don't want this in the the catch block? And the size check in the try block? It might look cleaner? Just a matter of style though so happy with this PR either way.

Reply:
It has to be like this to avoid duplication, since `database.getFileByCommitSHA(....)` can also return `null` which needs to be registered as the same type of content problem.